axelor-account/src/main/java/com/axelor/apps/account/service/AccountingCutOffServiceImpl.java

```diff
@@ -360,14 +360,15 @@ protected void generateMoveLinesFromMove(
         // Check if move line already exists with that account
         if (cutOffMoveLineMap.containsKey(moveLineAccount)) {
           cutOffMoveLine = cutOffMoveLineMap.get(moveLineAccount);
-          cutOffMoveLine.setCurrencyAmount(
-              cutOffMoveLine.getCurrencyAmount().add(amountInCurrency));
+          BigDecimal currencyAmount = cutOffMoveLine.getCurrencyAmount().add(amountInCurrency);
           if (isReverse
               != (accountingCutOffTypeSelect
                   == AccountingBatchRepository.ACCOUNTING_CUT_OFF_TYPE_DEFERRED_INCOMES)) {
             cutOffMoveLine.setDebit(cutOffMoveLine.getDebit().add(convertedAmount));
+            cutOffMoveLine.setCurrencyAmount(currencyAmount.abs());
           } else {
             cutOffMoveLine.setCredit(cutOffMoveLine.getCredit().add(convertedAmount));
+            cutOffMoveLine.setCurrencyAmount(currencyAmount.negate());
```

Comment:
the output of negate() will be positive if currencyAmount is negative, so I'm not sure if it will work in every cases...